### PR TITLE
ARM64: dts: ast2700-evb: Enable Uart/SPI/soc/pinctrl/irqchip/net/wdt drivers changes

### PR DESCRIPTION
**Pull request by ashok2ku**
Adding initial Commit to support Aspeed 2700 board

### CODE REVIEW

**Comment by ojayanth** [comment]
arch/arm64/boot/dts/aspeed/Makefile  related evb dtb related changes  missing ?

It will be good to add additional information's for future reference.
Linux shell support for ast2700 eval board.
Key- features enabled : uart/spi driver updates and minimum set of features required to boot eval board on shell.
Linux Kernel base : 6.6.36 version. 
Aspped tree Reference:
Verified : details.

**Comment by ashok2ku** [comment]
> arch/arm64/boot/dts/aspeed/Makefile related evb dtb related changes missing ?
> 
> It will be good to add additional information's for future reference. Linux shell support for ast2700 eval board. Key- features enabled : uart/spi driver updates and minimum set of features required to boot eval board on shell. Linux Kernel base : 6.6.36 version. Aspped tree Reference: Verified : details.

Added makefile and updated commit message

**Review by rajaganeshr — APPROVED** [review]

File: arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-marley.dts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+&pinctrl1 {
+	pinctrl_rgmii0_driving: rgmii0_driving {
+		pins = "C20", "C19", "A8", "R14", "A7", "P14",
+		       "D20", "A6", "B6", "N14", "B7", "B8";
+		drive-strength = <1>;
+	};
+};
+
+
```
Comment (ojayanth) [inline arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-marley.dts]: Remove extra line

File: arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-marley.dts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	flash@1 {
+		status = "okay";
+		m25p,fast-read;
+		label = "mpflash";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <4>;
+		spi-rx-bus-width = <4>;
+	};
+
```
Comment (ojayanth) [inline arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-marley.dts]: Remove extra line

File: arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-marley.dts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+	pinctrl-names = "default";
+
+	flash@0 {
+		status = "okay";
+		m25p,fast-read;
+		label = "bmc";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <4>;
+		spi-rx-bus-width = <4>;
+		#include "aspeed-flash-layout-128.dtsi"
```
Comment (ojayanth) [inline arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-marley.dts]: Usually #include aligns to node. , Should align to  line 81.

File: arch/arm64/boot/dts/aspeed/Makefile [hunk 4]
```diff
@@ -1,2 +1,3 @@
 dtb-$(CONFIG_ARCH_ASPEED) += \
```
Comment (ojayanth) [inline arch/arm64/boot/dts/aspeed/Makefile:1]: Comments Commit message :
ARM64: dts: aspeed:  Adding AMD Marley

Added base version of device tree file for AMD Marley platform based on
Aspeed 2700 SOC malta

Tested: Verified on Marley with Malta by  booting to shell.

File: arch/arm64/Kconfig.platforms [hunk 5]
```diff
@@ -33,6 +33,20 @@ config ARCH_APPLE
 	  This enables support for Apple's in-house ARM SoC family, starting
 	  with the Apple M1.
 
+config ARCH_ASPEED
```
Comment (ojayanth) [inline arch/arm64/Kconfig.platforms:36]: Commit message:

ARM64: dts: ast2700-evb: Adding  linux shell  boot support

DTS/Device tree patches from AspeedTech-BMC/linux/tree/aspeed-master-v6.6 branch
Linux Kernel base : 6.6.36 version.

Added minimum set of features required to boot Aspeed 2700 EVB board on shell
- uart/spi/soc/pinctrl/irqchip/net/wdt driver updates from AspeedTech-BMC
- clock/reset and 2700 related header file
- Added Aspeed tree -ast2700-evb.dts and dependent dts file
 
Tested: ast2700 evb machine boots to shell on evaluation board/QEMU
            OpenBMC Linux base Version: 6.6.36 
            QEMU: 02d9c38236c

**Review by mahkurap — APPROVED** [review]

**Review by mahkurap — APPROVED** [review]

**Review by ojayanth — APPROVED** [review]

**Pull request merged**